target/riscv/cheri-archspecific.h

```diff
@@ -56,7 +57,24 @@ cheri_tag_prot_clear_or_trap(CPUArchState *env, target_ulong va,
                              int cb, const cap_register_t* cbp,
                              int prot, uintptr_t retpc, target_ulong tag)
 {
-    // TODO: implement load-inhibit tlb flags
+#ifndef TARGET_RISCV32
```

Comment:
Since this doesn't use any architectecture-specific constants other than the trap number maybe we should move it to common and add an architecture-dependent `cheri_raise_load_cap_page_fault()` or similar?

The ifdef shouldn't be needed, since the flags just won't ever be set. Or do we care about this micro-optimization?

Reply:
I see it more as documenting that this code is dead code on RV32 rather than someone thinking there might be another way to trigger it. If everything is guarded by !RV32 then it's obvious (if a little cluttered). But if it becomes target-independent then that's less of a concern because there's more of an expectation that different targets might not do everything.

Reply:
We don't define the trap number at all for RV32, so the PAGE_LC_TRAP portion has to be conditional and Jessica had asked for the PAGE_LC_CLEAR bits to be as well earlier.  I do think we could make most of this function shared so long as we had some arch-specific function to raise the load_cap exception, but perhaps that could be done as a followup change?  The followup change might be the right time to move the #ifdef inside of the arch-specific function definition.

Reply:
A follow-up change sounds good.